### PR TITLE
Create afraser.md

### PR DESCRIPTION
**Pull request by afraser**

### CODE REVIEW

File: soapbox/people/afraser.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# [Adam's](https://github.com/afraser) Working-with-humans Soapbox
+
+The purpose of this repository is to document our engineering practices, and when you ask an engineer in the trenches of a project to talk about this sort of thing they might start talking about the stack they've chosen. A stack, even at a macro scale, is an implementation detail. The good stuff comes when we ask "why" and then ask it again and again.
+
+The answer always circles back to _people_.
```
Comment (theothermattm) [inline soapbox/people/afraser.md:5]: ❤️

File: soapbox/people/afraser.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+- Think hard about naming. 
+- Use sensible abstractions and try to write “idiomatic” code adhering to framework patterns.
+- Provide sensible error handling for developers and users.
+- Always consider worst-case performance.
+- Always have a single source of truth for application state.
+- Write unit tests.
+- Avoid hacks and “magic numbers”. If you can’t, leave notes in comments for future travellers.
+- Employ responsive styles and sensible css transitions.
+- Optimize for the intended platform and user: Phone or desktop? Consumer or professional?
+- Avoid files with over 200 lines of code
```
Comment (theothermattm) [inline soapbox/people/afraser.md:60]: I think this needs a little more explanation, I think I know what you're after but could use a little expansion

File: soapbox/people/afraser.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+Get decision makers on your side and/or find a wolf pack. Communicate with and become best friends with Design, UX, Product Owners, Architecture, DevOps, etc.
+
+## Pay attention to detail.
+Don’t get sloppy (slow code, bad naming, missed edge cases, etc.). Delight developers, designers and users. That said, don't let the details derail you from delivering on your promises.
+
+- Think hard about naming. 
+- Use sensible abstractions and try to write “idiomatic” code adhering to framework patterns.
+- Provide sensible error handling for developers and users.
+- Always consider worst-case performance.
+- Always have a single source of truth for application state.
```
Comment (theothermattm) [inline soapbox/people/afraser.md:55]: A little more explanation?

**Comment by theothermattm** [comment]
This is so good... Can you put a new section and link in the main README.md please?

File: soapbox/people/afraser.md (hunk 3 above)
Comment (afraser) [reply soapbox/people/afraser.md:55]: Not sure what to add here. SSOT is a pretty widely understood concept I think, is the term "application state" making it more confusing?

File: soapbox/people/afraser.md (hunk 2 above)
Comment (afraser) [reply soapbox/people/afraser.md]: I just went ahead and removed it... these notes are mostly aimed at senior devs and I'm probably trying to state the obvious.

**Comment by afraser** [comment]
@theothermattm Added the readme link.

File: soapbox/people/afraser.md (hunk 3 above)
Comment (theothermattm) [reply soapbox/people/afraser.md:59]: To me as a back end dev the single source of truth is the API and/or database.  But I'm not sure if that's what you mean?  I think I know what you're getting at but if I were more junior I probably wouldn't.

File: README.md [hunk 4]
```diff
@@ -101,24 +103,16 @@ Oh, but don't say the "F" word unless the client does first, please 😜
 
 ## What has worked for us in the past
 
-We've found the following things work really well working with clients:
+Establishing trust with clients can make the difference between getting the job done and exceeding expectations, [afraser](https://github.com/afraser) has some opinions on this stuff, but here's a high-level:
```
Comment (theothermattm) [inline README.md:106]: I would link to your soapbox instead of your profile.

File: README.md [hunk 5]
```diff
@@ -12,7 +12,9 @@
     - [What deployment platform will you use?](#what-deployment-platform-will-you-use)
     - [Other questions to consider](#other-questions-to-consider)
 - [Soapboxes](#soapboxes)
+  - [Angular](#angular)
   - [General Software Development Best Practices](#general-software-development-best-practices)
+  - [Java](#java)
   - [Node.js](#nodejs)
```
Comment (theothermattm) [inline README.md:18]: Is your soapbox missing here?

File: README.md [hunk 6]
```diff
@@ -101,24 +103,16 @@ Oh, but don't say the "F" word unless the client does first, please 😜
...
-- Build it, then show and tell.
-  - New ideas are much more likely to get pulled in if they're already "mostly done."
-- Make developers' lives easier.
-  - Automate things. Improve performance. Find pain points and fix them.
-- Transcend your team.
-  - Communicate and become best friends with Design, UX, Product Owners, Architecture, DevOps... Etc.
-- Pay attention to detail.
-  - Don’t get sloppy (slow code, bad naming, missed edge cases, etc.) .
-  - Delight developers, designers and users.
+- Deliver on promises.
```
Comment (theothermattm) [inline README.md:109]: I like this. Would it make sense to say:

> Set expectations.  Deliver on promises.

File: README.md [hunk 7]
```diff
@@ -101,24 +103,16 @@ Oh, but don't say the "F" word unless the client does first, please 😜
...
+- Speak from experience (and in terms of what makes people happy or unhappy)
```
Comment (theothermattm) [inline README.md:110]: Present ideas frequently is gone.  I would add it back and maybe even add to it:

> Present ideas frequently and get feedback.  Get decisionmakers on your side.

File: soapbox/people/afraser.md (hunk 3 above)
Comment (afraser) [reply soapbox/people/afraser.md:59]: This applies on the backend as well. eg: you shouldn't have two tables that contain the same column data unless we're talking about a foreign key.

Comment (theothermattm) [reply soapbox/people/afraser.md:59]: Got it.  How about something like this?

> Don't store data in two places, always have a single source of truth for application state.

File: README.md (hunk 6 above)
Comment (afraser) [reply README.md:109]: love it!

File: soapbox/people/afraser.md (hunk 3 above)
Comment (afraser) [reply soapbox/people/afraser.md:59]: How about this: 

Always have a [SSOT](https://en.wikipedia.org/wiki/Single_source_of_truth).

...trying hard to keep this as succinct as possible.

Comment (afraser) [reply soapbox/people/afraser.md:59]: Also, while we're on this, do you have any points you'd add to this and/or the "attention to detail" section. I have a very "front-endy" point of view 😝

Comment (afraser) [reply soapbox/people/afraser.md:59]: Also I'm currently updating this list to be collapsible because it takes up a disproportionate amount of space in the article to it's relevance.

Comment (theothermattm) [reply soapbox/people/afraser.md:59]: I like the update! Links are great.

Comment (theothermattm) [reply soapbox/people/afraser.md:59]: Re: more stuff for backend... Maybe "Make it easy for developers to get the application running locally"?

File: README.md (hunk 4 above)
Comment (afraser) [reply README.md]: oops!

File: README.md (hunk 7 above)
Comment (afraser) [reply README.md]: I actually rolled the decisionmaker bit into "Transcend your team." ... I guess I was hesitant about this one because it didn't feel as put-together of an idea. I'll try and fit it back in.

File: soapbox/people/afraser.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+## Speak in terms of real users and use cases.
+
+The better you understand who is going to use the product, the better you will understand how it should be built. This is just one case where it pays to know people outside of the engineering team.
+
+## Pay attention to detail.
+
+Delight developers, designers and users. Putting the work in here pays dividends in trust. That said, don't let this derail you from delivering on your promises. And this goes without saying but don’t get sloppy (bad naming, missed edge cases, slow code, etc.).
+
+<details>
```
Comment (theothermattm) [inline soapbox/people/afraser.md:56]: Did not know about this, cool!

File: README.md [hunk 9]
```diff
@@ -101,24 +103,18 @@ Oh, but don't say the "F" word unless the client does first, please 😜
 
 ## What has worked for us in the past
 
-We've found the following things work really well working with clients:
+[Establishing trust with clients](soapbox/people/afraser.md) can make the difference between getting the job done and exceeding expectations, [afraser](https://github.com/afraser) has some opinions on this stuff, but here's a high-level:
```
Comment (theothermattm) [inline README.md:106]: I would reword to this 
```suggestion
Establishing trust with clients can make the difference between getting the job done and exceeding expectations, [Adam F's soapbox on this has good suggestions](soapbox/people/afraser.md) but here's a high-level overview:
```

File: README.md [hunk 10]
```diff
@@ -101,24 +103,18 @@ Oh, but don't say the "F" word unless the client does first, please 😜
...
-  - Don’t get sloppy (slow code, bad naming, missed edge cases, etc.) .
-  - Delight developers, designers and users.
+- Set expectations. Deliver on promises.
+- Speak in terms of what makes people happy or unhappy.
+- Big ideas are more likely to be entertained if they're already built. Build it, then show & tell.
+- Make developers' lives easier through automation etc.
+- Transcend your team by communicating with Design, UX, etc.
+- Speak in terms of real users and use cases.
+- Pay attention to detail. Delight developers, designers and users.
+- Know when and how much to push back.
```
Comment (theothermattm) [inline README.md:116]: I like this.  But what about:

`Know when and how much to push back. Pick your battles and always _be tactful and respectful_.`

Comment (afraser) [reply README.md:116]: I feel like the second bit is redundant (pick your battles is the "when", I'll add the bit about tact to the soapbox though)

**Review by theothermattm — APPROVED** [review]
So good! SHIP IT
